bigtable/client/internal/table_admin.cc

```diff
@@ -188,6 +188,33 @@ bool TableAdmin::CheckConsistency(
   return response.consistent();
 }
 
+bool TableAdmin::WaitForConsistencyCheckImpl(
+    bigtable::TableId const& table_id,
+    bigtable::ConsistencyToken const& consistency_token,
+    std::unique_ptr<bigtable::PollingPolicy> polling_policy) {
+  btproto::CheckConsistencyRequest request;
+  request.set_name(TableName(table_id.get()));
+  request.set_consistency_token(consistency_token.get());
+  MetadataUpdatePolicy metadata_update_policy(
+      instance_name(), MetadataParamTypes::NAME, table_id.get());
+
+  grpc::Status status;
+  do {
+    auto response = ClientUtils::MakeCall(
+        *client_, rpc_retry_policy_->clone(), rpc_backoff_policy_->clone(),
+        metadata_update_policy, &AdminClient::CheckConsistency, request,
+        "CheckConsistency", status, true);
+
+    if (status.ok()) {
+      return true;
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
The sleeping time probably should come from the polling policy.

Reply:
In fact, we should consider using the ExponentialBackoffPolicy implementation, but maybe not the interface.